internal/pkg/bashexec/bashexec.go

```diff
@@ -0,0 +1,83 @@
+package bashexec
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+// BashRunner is a wrapper for executing in-memory bash scripts
+type BashRunner struct {
+	name string
+	cmd  *exec.Cmd
+}
+
+// NewBashRunner creates a bash executor from in-memory shell script.
+func NewBashRunner(name, src string, args ...string) (*BashRunner, error) {
+	// This will be proxied to fd 3
+	f, err := os.CreateTemp("", name)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := io.Copy(f, strings.NewReader(src)); err != nil {
+		return nil, err
+	}
+	if err := os.Remove(f.Name()); err != nil {
+		return nil, err
+	}
+
+	fullargs := append([]string{"-c", ". /proc/self/fd/3", name}, args...)
+	cmd := exec.Command("/bin/bash", fullargs...)
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGTERM,
+	}
+	cmd.Stdin = os.Stdin
+	cmd.ExtraFiles = append(cmd.ExtraFiles, f)
+
+	return &BashRunner{
+		name: name,
+		cmd:  cmd,
+	}, nil
+}
+
+// Exec synchronously spawns the child process, passing stdin/stdout/stderr directly.
+func (r *BashRunner) Exec() error {
+	r.cmd.Stdin = os.Stdin
+	r.cmd.Stdout = os.Stdout
+	r.cmd.Stderr = os.Stderr
+	err := r.cmd.Run()
+	if err != nil {
+		return fmt.Errorf("failed to execute internal script %s: %w", r.name, err)
+	}
+	return nil
+}
+
+// Run spawns the script, gathering stdout/stderr into a buffer that is displayed only on error.
+func (r *BashRunner) Run() error {
```

Comment:
Might be marginally helpful if we disambiguate this `Run()` with the one below (line 77). The function signature's are different so the compiler knows (obviously), but could trip up someone reading the code.